modules/local/rename_contigs.nf

```diff
@@ -0,0 +1,30 @@
+process RENAME_CONTIGS {
+    tag "${meta}"  // Directly use meta as it's already an ID
+    label 'process_low'
+
+    conda "bioconda::seqkit=2.3.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqkit:2.3.1--h9ee0642_0' :
+        'quay.io/biocontainers/seqkit:2.3.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(contigs)
+
+    output:
+    tuple val(meta), path("${meta}_renamed_contigs.fa.gz"), emit: renamed_contigs
+    path "versions.yml", emit: versions
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta}"  // Use meta directly as the prefix
```

Comment:
```suggestion
    def prefix = task.ext.prefix ?: "${meta.id}"
```
I think, again, that it would be better to stick to the convention of having a full `meta` object.